File: services/src/main/java/com/senacor/code/fullstack/chat/InitDatabaseOnStartupRunner.java
package com.senacor.code.fullstack.chat;

import com.senacor.code.fullstack.chat.domain.Channel;
import com.senacor.code.fullstack.chat.domain.ChatMessage;
import com.senacor.code.fullstack.chat.repository.ChannelRepository;
import com.senacor.code.fullstack.chat.repository.ChatMessageRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

/**
 * Class to fill our embedded Mongo DB with some test data.
 */
@Component
public class InitDatabaseOnStartupRunner implements CommandLineRunner {

    private static final Logger LOGGER = LoggerFactory.getLogger(InitDatabaseOnStartupRunner.class);

    @Autowired
    private ChannelRepository channelRepository;

    @Autowired
    private ChatMessageRepository chatMessageRepository;

    @Override
    public void run(String... args) throws Exception {
        // create some channels
        channelRepository.save(new Channel("general", "general"));
        channelRepository.save(new Channel("dev", "dev"));
        channelRepository.save(new Channel("humor", "humor"));

        // fetch all channels
        LOGGER.info("Channels in Mongo DB:");
        for (Channel channel : channelRepository.findAll()) {
            LOGGER.info("    " + channel);
        }

        // create some messages
        chatMessageRepository.save(new ChatMessage("general", "sender@test.de", "Hello"));
        chatMessageRepository.save(new ChatMessage("general", "sender@test.de", "World!"));
        chatMessageRepository.save(new ChatMessage("dev", "sender@test.de", "Did you ever tried Spring Boot 2.0.0?"));

        // fetch all channels
        LOGGER.info("ChatMessages in Mongo DB:");
        for (ChatMessage msg : chatMessageRepository.findAll()) {
            LOGGER.info("    " + msg);
        }

    }
}

File: services/src/main/kotlin/com/senacor/code/fullstack/chat/InitDatabaseOnStartupRunner.kt
package com.senacor.code.fullstack.chat

import com.senacor.code.fullstack.chat.channel.Channel
import com.senacor.code.fullstack.chat.channel.ChannelRepository
import com.senacor.code.fullstack.chat.message.ChatMessage
import com.senacor.code.fullstack.chat.message.ChatMessageRepository
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.CommandLineRunner
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Component
@Profile("!production")
class InitDatabaseOnStartupRunner : CommandLineRunner {

    private val log = LoggerFactory.getLogger(InitDatabaseOnStartupRunner::class.java)

    @Autowired
    private lateinit var channelRepository: ChannelRepository

    @Autowired
    private lateinit var chatMessageRepository: ChatMessageRepository

    @Throws(Exception::class)
    override fun run(vararg args: String) {
        // create some channels
        channelRepository.save(Channel("general", "general"))
        channelRepository.save(Channel("dev", "dev"))
        channelRepository.save(Channel("humor", "humor"))

        // create some messages
        chatMessageRepository.save(ChatMessage("dev", "julia@test.de", "Hello"))
        chatMessageRepository.save(ChatMessage("dev", "julia@test.de", "World!"))
        chatMessageRepository.save(ChatMessage("general", "max@test.de", "Have fun!"))
        chatMessageRepository.save(ChatMessage("general", "tcuje@freenet.de", "Hello my name is Thomas"))

        // fetch all channels
        log.info("Channels in Mongo DB:")
        for (msg in channelRepository.findAll()) {
            log.info("    $msg")
        }

        // fetch all messages
        log.info("Messages in Mongo DB:")
        for (msg in chatMessageRepository.findAll()) {
            log.info("    $msg")
        }
    }
}